docs/proposal-state-names.md

```diff
@@ -0,0 +1,57 @@
+# Proposal
+
+## Rationalise Task State, Output, and Trigger Names
+
+REF: https://github.com/cylc/cylc-ui/pulls#issuecomment-521121132
+
+Cylc has gradually accrued new task states and associated triggers and outputs
+over the years, and the resulting names are variously ambiguous, confusing, or
+grammatically inconsistent.
+
+### Task (and Job) States
+
+| old             | new :question:    | meaning | comment |
+|----             |----               | ---     | ---     | 
+| runahead        | -                 | beyond "max active cycle points" | (not needed in spawn-on-demand future) |
+| waiting         | -                 | waiting for prerequisites to be met |
+| queued          | limited           | held back by a size-limited internal queue | **queued** gets confused with "submitted" (to a batch queue)! |
+| ready           | submitting        | queued to the subprocess pool for job submission | | 
+| expired         | -                 | waiting for too long, will never be submitted | | 
+| submitted       | -                 | job submitted to batch system for execution | | 
+| submit-failed   | submission-failed | job submission failed, did not execute | | 
+| submit-retrying | submission-retry  | job submission failed, will try again later | retry**ing** suggests already re-submitted |
+| running         | executing         | job execution commenced | | 
+| succeeded       | execution-succeeded | job execution completed successfully | | 
+| failed          | execution-failed  | job execution completed, but failed | | 
+| retrying        | execution-retry   | job execution failed, will try again later | retry**ing** suggests already re-executing | 
+| held | - | held back even if ready to submit | (not actually a task state in Cylc 8) |
```

Comment:
Can be removed from this list, no longer a task status, now an attribute of the task state, i.e. orthogonal to and at the same level as the task status.

Reply:
Yep.